### ISSUE TITLE
SINGULARITY_DOCKER_USERNAME not export in bootstrap

### ISSUE THREAD

**Issue by vsoch** [apptainer/singularity#726]
## Version of Singularity:

2.3

### Expected behavior

Using bootstrap to create a container should find the environment variable for `SINGULARITY_DOCKER_USERNAME` akin to how import does.

### Actual behavior

It is not found. Likely the export of the variable is not done in the bootstrap flow as it is done for import.

### Steps to reproduce behavior

This is from one of our users in the list

```
export SINGULARITY_DOCKER_USERNAME=toto
wget https://raw.githubusercontent.com/singularityware/singularity/master/examples/docker/Singularity
singularity create --force --size 2048 busybox.img
sudo -E singularity -vvv bootstrap busybox.img Singularity

VERBOSE2 SINGULARITY_DOCKER_USERNAME not defined (None)
VERBOSE2 SINGULARITY_DOCKER_PASSWORD found
```

**Comment by gmkurtzer** [comment]
Ahh, yes. The `bootstrap` command cleans the environment, so we just have to add an exception (or rather a reset) of this variable. So I just need to make sure that `SINGULARITY_DOCKER_USERNAME` and `SINGULARITY_DOCKER_PASSWORD` are reset in bootstrap, correct?

**Comment by vsoch** [comment]
correct!

**Comment by gmkurtzer** [comment]
Excellent, I got it all ready for testing in the development branch.

**Comment by Trophime** [comment]
Could we backport the patch to fix this issue on 2.3 series?

**Comment by dtrudg** [comment]
Closing, as this issue was fixed in merged PRs. Please let us know if it's still an issue on current Singularity release. Thanks.
